python/cuml/cuml/accel/tests/scikit-learn/xfail-list.yaml

```diff
@@ -1318,3 +1318,7 @@
   marker: cuml-accel_kmeans_fails_with_attribute_error_memmap_ptr
   tests:
   - "sklearn.cluster.tests.test_k_means::test_predict_does_not_change_cluster_centers[None]"
+- reason: "The signs of the principal components are swapped with cuml.accel (insignificant deviation)"
+  condition: "cuda-python >= 12.9"
```

Comment:
How does this `cuda-python` check work? Is it checking the package `cuda-python`'s version?

If so, it is worth noting that `cuda-python` version `12.9` will be picked up and used anywhere CUDA 12.0+ works

So if our goal is to skip this test only when using CUDA 12.9, we might want to do something different. One option we could try is checking the cuBLAS or cuSOLVER versions if this relates to the behavior of those libraries. There may be other options to try depending on how we would like to filter here

Reply:
Considering that this is only relevant within our test environment, I thought this would be a sufficient check. At least right now it seems to be picking up the right thing. I think the correct version is enforced through the `cuda-version` metapackage:

https://github.com/rapidsai/cuml/blob/e433dbc3665b2b3e076d31a88f53f12b86cadaf0/conda/environments/all_cuda-128_arch-x86_64.yaml#L16

Unfortunately that's not within the distribution list.

Reply:
It's worth noting that we could also drop the filter entirely and just xfail indiscriminately, but in general, I'd rather be as conservative as possible with the xfails.

Reply:
Can understand the desire to check and be conservative on skips

So IIUC this is checking `pip list` or how is this check implemented?

Reply:
Kind of. It obtains the version from the [metadata from the installed distribution](https://docs.python.org/3/library/importlib.metadata.html#importlib.metadata.version) and then evaluates the requirement string with `packaging`'s [`Requirement`](https://packaging.pypa.io/en/stable/requirements.html) class:

https://github.com/rapidsai/cuml/blob/e433dbc3665b2b3e076d31a88f53f12b86cadaf0/python/cuml/cuml/accel/pytest_plugin.py#L63-L68